### ISSUE TITLE
🐛 BUG: CSS imported have higher priority than CSS on .astro file

### ISSUE THREAD

**Issue by Eloi-Perez** [withastro/astro#3357]
### What version of `astro` are you using?

1.0.0-beta.27

### Are you using an SSR adapter? If so, which one?

no

### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

When I import CSS like `import '../styles/style.css';` or CSS contained in other .astro file like `import Layout from '../layouts/Layout.astro';` This imported CSS as priority over the CSS inside the `<style>` on a .astro file.
This way, an external CSS rule with the same specificity will overwrite the one on `<style is:global>`
And you could say, use scoped `<style>`... but certain elements are not affected by the `:global()` or `is:global`, like `*`, `html` or `body`.
So if I create a CSS rule for `body` on an external file it would be hard to change it unless I use `!important`
We should not rely on specificity for this, the internal CSS should always be written at the bottom of the file.

To sum up, the CSS file (asset.xxx.css) from `astro build` and the temporal CSS file from `astro dev`. IMA have the content in the wrong order.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-ndeuc4-qbjgyn?file=src%2Fpages%2Findex.astro,src%2Fpages%2Fpage2.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by natemoo-re** [comment]
Thanks for opening an issue! You're right that our specificity is a bit wonky and difficult to control right now.

This problem is something we've discussed in the past but never came to a resolution on. I'm tagging this issue to be reviewed and discussed by the core team.

Related: we have an open PR to ensure that Astro scoped styles use `:where()` to avoid adding unintended specificity https://github.com/withastro/compiler/pull/330. Would that fix your problem or do you still think the output order has to be adjusted?

**Comment by Eloi-Perez** [comment]
Hi @natemoo-re , I've been using the `@layer` CSS property as as a way to get around it.
But since I recently updated to the last Astro version (1.0.0-beta.42). Now `astro dev` and `astro build` order the CSS in different ways making it very inconvenient to write CSS as I may get different results on build time.

**Comment by filipwiniarski** [comment]
+1 - Global styles imported in components land before TailwindCSS base styles, losing specificity battle in the result.

**Comment by ghostdevv** [comment]
really hoping a solution appears :pray:

**Comment by Hotell** [comment]
this happens in 1.0 stable as well

**Comment by tudorpopams** [comment]
> Hi @natemoo-re , I've been using the `@layer` CSS property as as a way to get around it. But since I recently updated to the last Astro version (1.0.0-beta.42 from 1.0.0-beta.27). Now `astro dev` and `astro build` order the CSS in different ways making it very inconvenient to write CSS as I may get different results on build time.

The `astro dev` vs `astro build` CSS order issue is still happening in **the stable release (v1.0.5)**. This creates a big predictability and regression issue on production environments and the workarounds are very hacky. Is there any chance for this to get a higher priority?

**Comment by Eloi-Perez** [comment]
astro@1.0.2
I've seen that now mentioned in the documentation the possibility to "Load a static stylesheet via “link” tags" so I've tested it, at the moment, the order of loading is: 

- Layout.astro `<head><link rel="stylesheet" href="/styles/test.css" /></head>` (CSS on public folder)
- index.astro `<style></style>`
- Layout.astro `<style is:global></style>`
- Layout.astro `import '../styles/test.css';`
- index.astro `import '../styles/test.css';`

Same results using `astro dev` and `astro preview`

So using "static stylesheet via “link” tags" skips the normal CSS processing, bundling and optimizations BUT, you could write your globals there and you will have the possibility to overwrite them with `<style>` inside `.astro` files.

**Comment by pikeas** [comment]
Please consider revising CSS load order to:

1) Layout.astro `<head><link rel="stylesheet" href="..." /></head>`
2) Layout.astro `import '...'`
3) Layout.astro `<style is:global></style>`
4) index.astro `import '...`
5) index.astro `<style></style>`

**Comment by matthewp** [comment]
Hey everyone, we just pushed at a fix for this that will be released in 1.4.0. Let me clarify the ordering as it's intended so that we're all on the same page. Let's say you have this:

```astro
---
import '../styles/base.css';
import '../styles/global.css';
---
<html>
  <head>
    <link rel="stylesheet" href="/another.css">

    <style>
      body {
        background: limegreen;
      } 
    </style>
  </head>
</html>
```

The ordering you should see is:

```html
<link rel="stylesheet" href="/another.css">

<link rel="stylesheet" href="/styles/base.css">
<link rel="stylesheet" href="/styles/global.css">

<!-- this is your <style> -->
<link rel="stylesheet" href="/src/pages/index.astro?type=style...">
```

The change made is that your `<style>` always comes *after* imported styles. This fixes the bug that people have been seeing.

Here's a couple of things to remember:

1. We **do not** touch the `<link>` you manually add in any way. We are effectively unaware of them.
2. We always place the `<link>`s that we add at the bottom of the `<head>`.
3. You can control the ordering by controlling the order that you import things. In the above example, if you want `base.css` to be lower in the HTML order, then import it after `global.css`. This is also true of components. If you want to prioritize their styles over `global.css`, then make `global.css` the first thing you import.

I hope that clears things up! There's still likely some issues to address. Code-splitting in particular makes preserving order very hard, so built-order is more likely to not follow the above rules. Please still report anything that you find.

**Comment by Eloi-Perez** [comment]
Thank you, I just tested 1.4.0 and it works as intended!
